File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/StackTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class StackTransformer extends ABaseTransformer {

	@Override
	protected void onTransform(final View view, final float position) {
		view.setTranslationX(position < 0 ? 0F : -view.getWidth() * position);
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/StackTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View
import me.shkschneider.skeleton.extensions.then

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class StackTransformer : ABaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        page.translationX = (position < 0) then 1.toFloat() ?: -page.width * position
    }

}
